### PR TITLE
Fix some dummy label matches

### PR DESCRIPTION
**Pull request by tom-overton**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

In my PR for Obj_Etcetera, I fixed a match that involved a dummy label by using a scoped pad. I found out that a few other dummy labels in the codebase can be fixed the same way:

- For the actors, I'm pretty sure all of them are "true" matches.
- For game.c, I'm not sure if either of the things I changed are a "true" match or not, but they get rid of the dummy labels. If people don't like it, I'm happy to remove it from this PR.
- `DmaMgr_Start` in z_std_dma.c: I found that you can remove the dummy labels just by replacing them with `{ s32 pad }`. However, it looked so bad I didn't bother to include it in this PR.

### CODE REVIEW

File: src/code/game.c [hunk 1]
```diff
@@ -89,8 +89,11 @@ void GameState_Draw(GameState* gamestate, GraphicsContext* gfxCtx) {
     Graph_BranchDlist(_polyOpa, nextDisplayList);
     gfxCtx->polyOpa.p = nextDisplayList;
 
-lblUnk:; // Label prevents reordering, if(1) around the above block don't seem to help unlike in OoT
-    func_800E9F78(gfxCtx);
+    // Block prevents reordering, if(1) around the above block don't seem to help unlike in OoT
+    {
+        s32 pad;
+        func_800E9F78(gfxCtx);
```
Comment (engineer124) [inline src/code/game.c:95]:
```suggestion
        s32 pad;
        
        func_800E9F78(gfxCtx);
```

File: src/overlays/actors/ovl_En_Clear_Tag/z_en_clear_tag.c [hunk 2]
```diff
@@ -244,6 +244,7 @@ void EnClearTag_CreateLightRayEffect(EnClearTag* this, Vec3f* position, Vec3f* v
     // Look for an available effect to allocate a light ray effect to.
     for (i = 0; i < ARRAY_COUNT(this->effect) - 1; i++, effect++) {
         if (effect->type == CLEAR_TAG_EFFECT_AVAILABLE) {
+            s32 pad;
             effect->type = CLEAR_TAG_EFFECT_LIGHT_RAYS;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Clear_Tag/z_en_clear_tag.c:248]:
```suggestion
            s32 pad;
            
            effect->type = CLEAR_TAG_EFFECT_LIGHT_RAYS;
```

File: src/overlays/actors/ovl_En_In/z_en_in.c [hunk 3]
```diff
@@ -500,14 +500,14 @@ u16 func_808F3DD4(GlobalContext* globalCtx, EnIn* this, u32 arg2) {
     u16 textId = 0;
 
     if (Player_GetMask(globalCtx) == PLAYER_MASK_CIRCUS_LEADERS_MASK) {
+        s32 pad;
         if (!(gSaveContext.weekEventReg[63] & 0x40)) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_In/z_en_in.c:504]:
```suggestion
        s32 pad;
        
        if (!(gSaveContext.weekEventReg[63] & 0x40)) {
```

File: src/overlays/actors/ovl_En_In/z_en_in.c [hunk 4]
```diff
@@ -1259,9 +1259,9 @@ s32 func_808F5728(GlobalContext* globalCtx, EnIn* this, s32 arg2, s32* arg3) {
         return 1;
     }
     if (*arg3 == 1) {
+        s32 pad;
         player = PLAYER;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_In/z_en_in.c:1263]:
```suggestion
        s32 pad;
        
        player = PLAYER;
```

File: src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c [hunk 5]
```diff
@@ -2681,6 +2681,7 @@ void func_80B48AD4(EnInvadepoh* this, GlobalContext* globalCtx) {
     }
 
     if (this->rand == 0) {
+        s32 pad;
         if ((this->actor.xzDistToPlayer < 350.0f) && ((globalCtx->gameplayFrames & 0x60) != 0)) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c:2685]:
```suggestion
        s32 pad;
        
        if ((this->actor.xzDistToPlayer < 350.0f) && ((globalCtx->gameplayFrames & 0x60) != 0)) {
```

File: src/overlays/actors/ovl_En_Minislime/z_en_minislime.c [hunk 6]
```diff
@@ -324,14 +324,14 @@ void EnMinislime_IceArrowDamage(EnMinislime* this, GlobalContext* globalCtx) {
             Math_Vec3f_Copy(&this->shakeRefPos, &this->actor.world.pos);
         } else if (this->frozenTimer > 0) {
             if ((this->frozenTimer < 20) || ((this->frozenTimer < 40) && ((this->frozenTimer % 2) != 0))) {
+                s32 pad;
                 invFrozenTimer = 1.0f / this->frozenTimer;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Minislime/z_en_minislime.c:328]:
```suggestion
                s32 pad;
                
                invFrozenTimer = 1.0f / this->frozenTimer;
```

File: src/overlays/actors/ovl_En_Railgibud/z_en_railgibud.c [hunk 7]
```diff
@@ -391,12 +391,12 @@ void func_80BA5E18(EnRailgibud* this, GlobalContext* globalCtx) {
 
         case 1:
             if (this->unk_3F2 == 20) {
+                s16 pad;
                 sp32 = player->ageProperties->unk_92 + 0x6805;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Railgibud/z_en_railgibud.c:395]:
```suggestion
                s16 pad;
                
                sp32 = player->ageProperties->unk_92 + 0x6805;
```

File: src/overlays/actors/ovl_Obj_Raillift/z_obj_raillift.c [hunk 8]
```diff
@@ -234,6 +234,7 @@ void ObjRaillift_Update(Actor* thisx, GlobalContext* globalCtx) {
         }
     }
     if (OBJRAILLIFT_SHOULD_REACT_TO_WEIGHT(thisx)) {
+        s32 pad;
         this->isWeightOnPrev = this->isWeightOn;
```
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Raillift/z_obj_raillift.c:238]:
```suggestion
        s32 pad;
        
        this->isWeightOnPrev = this->isWeightOn;
```

**Review by engineer124 — COMMENTED** [review]
Nice find! This will be helpful to know moving forward as well. Only suggestion is to make sure there's a space between variable declarations and code

File: src/code/game.c (hunk 1 above)
Comment (tom-overton) [reply src/code/game.c:95]: Whoops, I was so excited about this trick that I forgot to do this. I'll fix this (and all the others below), thanks

Comment (EllipticEllipsis) [reply src/code/game.c:95]: This function is also missing DISPS macros, from the look of it, which might also affect things. I think #89 fixes that, but I wouldn't wait for it.

Comment (tom-overton) [reply src/code/game.c:95]: I got it matching with the DISPS macros; I had to delete the unused variables. I also cleaned it up in the same way that #89 did, but I can revert that cleanup if you'd rather it go in with lucas's PR.

**Review by AngheloAlf — APPROVED** [review]

**Review by EllipticEllipsis — APPROVED** [review]

**Review by engineer124 — APPROVED** [review]

**Review by isghj5 — APPROVED** [review]
Good to finally know the real cause of the label issue.

Only thing bothering me is the name of the variable `pad`. Naming a stack variable pad makes sense when we need to align the stack, but that's not what this variable does. This new variable does something else and the variable name should better represent what its doing... but I can't think of a good suggestion to replace it with. Best thing I could think of is like `hiddenScopeTemp` but that's still weird.

**Comment by tom-overton** [comment]
The best thing I can think of is just a riff on yours, which is `requiredScopeTemp`. I wouldn't necessarily call them "hidden" since we can sort of confirm that *something* like it existed in the original code (if they didn't, then we wouldn't have needed the dummy labels to begin with, and we wouldn't need these variables now). But since they are "required" to match, I figured making that part of the name would be descriptive enough.

**Review by rylieb — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**